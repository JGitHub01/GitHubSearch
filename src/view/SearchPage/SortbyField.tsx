interface ISortbyFieldProps {
  sortby?: string
  onSortbyChange?: (sortby: string) => void
}
export default function SortbyField({ sortby, onSortbyChange }: ISortbyFieldProps) {
  return (
    <div className='github-search__sortby' data-testid='sortby-field'>
      <label htmlFor='sortby'>Sort by:</label>
      <select data-testid='sortby-list'
        name='sortby'
        value={sortby}
        onChange={e => {
          onSortbyChange && onSortbyChange(e.target.value);
        }}
      >
        <option value=''>Best match</option>
        <option value='stars'>Stars</option>
      </select>
    </div>
  );
}
