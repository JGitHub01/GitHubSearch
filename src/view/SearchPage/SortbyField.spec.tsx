import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SortbyField from './SortbyField';

describe('SortbyField', () => {
  it('Should render the default snapshot view', () => {
    const { getByTestId } = render(<SortbyField />);
    const sortByField = getByTestId('sortby-field');
    expect(sortByField).toMatchSnapshot();
  });

  it('Should render the view with a set sort', () => {
    const { getByTestId } = render(<SortbyField sortby="stars" />);
    const sortByList = getByTestId('sortby-list');
    expect(sortByList).toHaveValue("stars");
  });

  it('Should call onSortbyChange with the new sort when a new sort is selected', () => {
    const callback = jest.fn();
    const { getByText, getByTestId } = render(<SortbyField onSortbyChange={callback} />);
    const sortByList = getByTestId('sortby-list');
    const starsOption = getByText('Stars');
    userEvent.selectOptions(sortByList, starsOption);
    expect(callback).toBeCalledWith('stars');
  });
});
