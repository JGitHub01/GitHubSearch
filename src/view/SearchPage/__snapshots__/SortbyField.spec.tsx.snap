// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortbyField Should render the default snapshot view 1`] = `
<div
  class="github-search__sortby"
  data-testid="sortby-field"
>
  <label
    for="sortby"
  >
    Sort by:
  </label>
  <select
    data-testid="sortby-list"
    name="sortby"
  >
    <option
      value=""
    >
      Best match
    </option>
    <option
      value="stars"
    >
      Stars
    </option>
  </select>
</div>
`;
